Enhance Name external links

JoeCohen opened the pull request:
- Adds additional external search links to Name `About this taxon` panel
- Reorganizes the panel:
  - Adds a **Research Links** caption, indenting external search links under that caption.

While working on the NEMF checklist, I've been wishing I had links like this in MO. So I finally added them. Some (e.g. MushroomExpert) are also useful for beginners.
(I also have an idea about adding these and similar links to the Observation pages. I'll write that up as a discussion.)

JoeCohen committed:
Add GBIF link

JoeCohen committed:
Add iNat and NCBI Nucleotide

JoeCohen committed:
Add MushroomExpert name search link

JoeCohen committed:
Add link to MycoGuide

JoeCohen committed:
Add wikipedia link to Name page

JoeCohen committed:
Change ncbi_nucleotide_name to .._term

The more general name parallels what the search does at ncbi, and allows for clearer re0sue of our methods.

JoeCohen committed:
Improve IF web searches

- Uses DuckDuckGo instead of Google
- Uses web search instead of a link to  IF name search page
(so that MO user doesn't have to re-enter the Name)

JoeCohen committed:
Re-arrange About this taxon panel

Puts Research Links in its own column for clarity, better balance

JoeCohen committed:
Use DuckDuckGo for mushroomexpert search

JoeCohen committed:
Add conditional Ascomycete.org link

- Adds a link to a search in Ascomycete.org for Names classified as ascos

JoeCohen committed:
Update ascomycete link stuff

JoeCohen committed:
Remove translating space to +

The translation is unecessary. Something is auotmagically encodes the spaces as %20

JoeCohen committed:
Alphabetically order NamesHelper name- & term_tab methods

JoeCohen committed:
Alpha order ObJectLinkHelper search_url methods

JoeCohen committed:
Update method names of searches which use duckduckgo

For clarity

JoeCohen committed:
Inline disable StringConcatenation cop

Fixes an offense for a line where interpolation would be inferior

JoeCohen committed:
Fix Codeclimate line length offenses

JoeCohen commented on app/helpers/object_link_helper.rb:
```diff
@@ -60,15 +60,42 @@ def name_link(name, str = nil)
-  # IF lacks an entry point that includes the name to be searched.
-  def index_fungorum_basic_search_url
-    "http://www.indexfungorum.org/Names/Names.asp"
+  # Use web search because IF internal search uses js form rather than a url
```
@nimmolo: What do you think of the DuckDuckGo search results?
In particular, would other search engines yield a more useful result?
(DDG has a pop-up the first time, and may show some ads before the first real result. But other search engines also do something obnoxious.)

nimmolo replied:
@JoeCohen I tried Google's "advanced search" syntax which is a little different - and it also returned no results for your example! Good idea trying DDG, seems like the way to go.

nimmolo replied:
I would maybe consider the "tab" pattern for these, as in app/helpers/tabs. 

I've been using those to give links identifying classes for system tests. The pattern is an array: 
`if_link_tab = [link_text, url, html_options]` - there are examples that auto-generate a class name. 
then `link_to(*if_link_tab)` in the template.

JoeCohen replied:
@nimmolo: Credit for the DuckDuckGo idea goes to @mo-nathan. I would never have thought of it.
Bing also returns decent results, but includes a Chat thingy that's more active, and therefore mroe annoying, than DuckDuckGo's thingy.

JoeCohen replied:
@nimmolo: Re "tab" pattern. I think that's what I'm doing (and it sure makes it easier to write the code). 
Maybe I'm misunderstanding the pattern or your comment.

nimmolo replied:
Oh, yes. I missed the rest of what you were doing... 
I was looking at this on my phone and only caught the new "url" helpers in object_link_helper.

mo-nathan commented:
Should there also be a vanilla "Google Search"?  Should there be a way to select different search engines (Bing, DuckDuckGo, etc.)?

mo-nathan commented:
The searches for "group"s doesn't work well.  Compare /names/58294 vs. /names/3914.  Maybe we shouldn't include " group" in group searches?

I'm not familiar with MycoGuide and it doesn't seem to have info at the species or even genus level (searches for "Tricholoma equestre" and "Tricholoma" are not at all compelling).  Why did this make the cut?

mo-nathan reviewed (approved):
Overall looks good and is a great improvement!  I added a few suggestions after playing around with it bit.  I leave it to @JoeCohen to decide which of those suggestions are worth implementing.

JoeCohen committed:
Move GoogleImages link to Research Links

JoeCohen commented:
> I think the "Google Images" link should go under "Research Links". It seems weird where it is now on the left side of that panel just above "Occurrence Map".

@mo-nathan: You're right. I moved the link.

JoeCohen committed:
Remove link to MycoGuide

JoeCohen committed:
DRY test_show_name_species_with_icn_id

JoeCohen committed:
Test for Ascomycete and MyCoPortal links

JoeCohen commented:
@mo-nathan: TLDR: I removed the link to MyCoGuide.
Back story:
It's Patrick Leacock's (the [NAMA Voucher Committee](https://namyco.org/about/committe) Coordinator) site. It's useful for the NEMF name work.
I agree it's not complete or strong enough for present purposes. 
(Might be useful in the future for other purposes, e.g., for Observations near Chicago.)
>I'm not familiar with MycoGuide and it doesn't seem to have info at the species or even genus level (searches for "Tricholoma equestre" and "Tricholoma" are not at all compelling). Why did this make the cut?

JoeCohen committed:
More test improvements

- Fixes incorrect assertion re MyCoPortal
- Tests absence & presence of conditional Ascomycete link
- Warns about adding ascos to names.yml. (Things blew up when I added one.)

JoeCohen committed:
Add method Name#s_str

JoeCohen committed:
Fine-tune external search strings

- Uses name s.s. in most cases.
- Sometimes quotes the name.

JoeCohen committed:
Rename Name#s_str => sensu_stricto

For clarity and to avoid ambiguity

JoeCohen committed:
Align Research Links with Nomenclature rh column

JoeCohen committed:
Fix ine-length offesnes

JoeCohen committed:
Add vanilla Google search

JoeCohen committed:
Fix bug in NameTest#test_sensu_stricto

JoeCohen commented:
@mo-nathan: I added a vanilla Google Search. It would be cool to have a way to select different search engines, but I don't want to deal with that now. I think it would require one of:
- Adding individual links for a each of a bunch of engines. I don't like this at all because of the clutter.
- Something like a drop-down menu to select the engine. This is desireable in the long run. But I think this requires a form or js, neither of which I have time to deal with now. 
> Should there also be a vanilla "Google Search"? Should there be a way to select different search engines (Bing, DuckDuckGo, etc.)?

mo-nathan commented:
Thanks!  Looks good.  I wonder if there is someway to provide a link that uses the users preferred search engine.  I did a bit of search around and didn't find anything that seemed right.  I don't think the `<search>` tag or `<input type="search">` do what I'm thinking of.  There's probably some JavaScript thing that does something reasonable.
